experimental/xla_tensor.py

```diff
@@ -0,0 +1,108 @@
+import torch
+import torch_xla
+import torch_xla.core.xla_model as xm
+import logging
+import functools
+import time
+from collections import namedtuple
+
+debug = False
+SIMPLE_TEST = False
+
+class XLATensor(torch.Tensor):
+    def __new__(cls, data, **kwargs):
+        return torch.Tensor._make_subclass(cls, torch.as_tensor(data, dtype=torch.float32, **kwargs))
+
+    def __init__(self, data, **kwargs):
+        super().__init__()
+        # if debug: print('[__init__]')
+        self.t_ = torch.as_tensor(data, dtype=torch.float32, **kwargs)
+        self.xla_tensor_sizes_ = namedtuple("XLATensorSizes", "static_size dynamic_size")
+
+    def size(self):
+        # if debug: print('[size] ')
+        static_size = super(XLATensor, self).size()
+        dynamic_size = torch_xla._XLAC._get_xla_tensor_dimension_size(self.t_, 0)
+        xla_tensor_sizes = self.xla_tensor_sizes_(static_size, self.t_) #TODO: replace self.t_ with dynamic_size
+        return xla_tensor_sizes 
+        #DynamicTensorXLASize(static_size) #TODO: done when no dynamic shame
+
+    def expand(self, size):
+        # if debug: print('[expand]')
+        # if size.dynamic_size != None:
+        # result = self.t_.expand(size.static_size) 
+        result = torch_xla._XLAC._xla_dynamic_expand(self.t_, size.dynamic_size, size.static_size)
+        return XLATensor(result.tolist()) #Q: return XLATensor or torch.Tensor?
+        # return self.t_.expand(size.static_size)
+
+    def __repr__(self):
+        # if debug: print ('[__repr__]')
+        return "XLATensor:\n{}\n".format(self.t_)
+
+    @classmethod
+    def __torch_function__(cls, func, types, args=(), kwargs=None):
```

Comment:
This isn't doing anything nontrivial, so we can get rid of it. (If we do end up needing to override a torch namespace function, that will be a global tax everywhere, see https://github.com/pytorch/pytorch/issues/62888 cc @anjali411 but nb the issue is about `__torch_dispatch__` but the need here is for `__torch_function__`)

Reply:
It seems like `XLATensor` might not need to be a tensor subclass at all. I am not sure if the `size()` overload is expected to be in C++. If yes, then it still needs to be a tensor subclass and we should use `_make_wrapper_subclass` if the size needs to be used in C++ as defined above.

Also, @ezyang the overhead here won't be as bad because we don't call from C++ to python for ever single function, right?